src/pattern-library/util/jsx-to-string.tsx

```diff
@@ -15,7 +15,7 @@ function escapeQuotes(str: string) {
  * Format a component's name for display
  */
 function componentName(type: any) {
-  let name = typeof type === 'string' ? type : type.displayName ?? type.name;
+  let name = typeof type === 'string' ? type : (type.displayName ?? type.name);
```

Comment:
Starting with prettier 3.3.3, parenthesis are enforced in nullish-coalescing used as part of ternaries https://github.com/prettier/prettier/blob/main/CHANGELOG.md#add-parentheses-for-nullish-coalescing-in-ternary-16391-by-cdignam-segment